extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudParsingHelper.java

```diff
@@ -73,10 +73,16 @@ public static String clientId() throws ExtractionException, IOException {
         throw new ExtractionException("Couldn't extract client id");
     }
 
-    static boolean checkIfHardcodedClientIdIsValid(Downloader dl) throws IOException, ReCaptchaException {
+    static boolean checkIfHardcodedClientIdIsValid(Downloader dl) {
         final String apiUrl = "https://api.soundcloud.com/connect?client_id=" + HARDCODED_CLIENT_ID;
-        // Should return 200 to indicate that the client id is valid, a 401 is returned otherwise.
-        return dl.head(apiUrl).getResponseCode() == 200;
+        try {
+            // Should return 200 to indicate that the client id is valid, a 401 is returned otherwise.
+            return dl.head(apiUrl).getResponseCode() == 200;
+        } catch (Exception ignored) {
```

Comment:
It might be good to throw the ReCaptcha exception. What do you think?

Reply:
If I understand correctly this function only has to check if the current client_id is correct, and if it is not it should return false and let `clientId()` fetch another client_id. So it is ok to just return false